### ISSUE TITLE
Reproducing the results of bertflow in the original paper

### ISSUE THREAD

**Issue by fzp0424** [UKPLab/pytorch-bertflow#4]
Hi, thank you for your great work! I am grateful to your pytorch-bertflow framework and I am using it to reproduce original bertflow experiments. But **the result(SRCC on STS-B) is always lower than it is reported in the paper**. I guess there are some details I ignore when reproducing. 
`loss: -1.180362  [473600/551204] 
corrcoef_dev: 0.223951 
loss: -1.119098  [480000/551204]
corrcoef_dev: 0.223691
loss: -1.132908  [486400/551204]
corrcoef_dev: 0.224357
loss: -1.211618  [492800/551204]
corrcoef_dev: 0.225161`
I choose **WNLI as the training set and STS-B as the dev set**. As the result shown above, **SRCC is about 22**, which is quite low.

**Comment by fzp0424** [comment]
Here is my simple script. I keep **TransformerGlow**  and other parts the same as yours.
I have several questions: 

- Did you reproduce the results of bert-flow using this framework successfully? 
- What is your training set? Your batch_size and lr?
- Is the **loss** I shown above normal?
- Does **the position and parameter settings of the tokenizer** make a difference? 
...
My script is easy to read and I give some definition of my class. Sorry to bother you!

```
class TrainDataset(Dataset):
    def __init__(self, data, tokenizer, maxlen, transform=None, target_transform=None):
        self.data = data
        self.tokenizer = tokenizer
        self.maxlen = maxlen
        self.transform = transform
        self.target_transform = target_transform

    def text_to_id(self, source):
        #         print(source)
        sample = self.tokenizer(source, add_special_tokens=True, return_tensors='pt',
                                max_length=self.maxlen, padding='max_length', truncation=True)
        return sample

    def __len__(self):
        return len(self.data)

    def __getitem__(self, idx):
        return self.text_to_id(self.data[idx])
    
def compute_corrcoef(x, y):
    '''calculate SRCC
    '''
    return scipy.stats.spearmanr(x, y).correlation


def eval_srcc(data, model):
    target_idxs, source_idxs, label_list = data.get_data()
    with torch.no_grad():
        target_input_ids = target_idxs['input_ids'].to(device)
        target_attention_mask = target_idxs['attention_mask'].to(device)
        target_token_type_ids = target_idxs['token_type_ids'].to(device)
        target_pred = model(
            target_input_ids, target_attention_mask)

        source_input_ids = source_idxs['input_ids'].to(device)
        source_attention_mask = source_idxs['attention_mask'].to(device)
        source_token_type_ids = source_idxs['token_type_ids'].to(device)
        source_pred = model(
            source_input_ids, source_attention_mask)

        similarity_list = F.cosine_similarity(target_pred, source_pred)
#         print(similarity_list[0])
        similarity_list = similarity_list.cpu().numpy()
        label_list = np.array(label_list)
        corrcoef = compute_corrcoef(label_list, similarity_list)

    return corrcoef

# Training parameters
train_batch_size = 64
num_epochs = 1
max_seq_length = 64

model_name_or_path = 'bert-base-uncased'

# Here we define our model
tokenizer = AutoTokenizer.from_pretrained(model_name_or_path)
training_data = TrainDataset(target_sample,tokenizer,max_seq_length)
train_dataloader = DataLoader(
    training_data, batch_size=train_batch_size, shuffle=True, drop_last=True)
dev_data = TestDataset(dev_samples, tokenizer, max_seq_length)

# model
encoder = TransformerGlow(model_name_or_path, pooling='first-last-avg').to(device)


# optimizer
no_decay = ["bias", "LayerNorm.weight"]
optimizer_grouped_parameters = [
    {
        "params": [p for n, p in encoder.glow.named_parameters()
                   if not any(nd in n for nd in no_decay)],  # Note only the parameters within diffusion will be updated and the Encoder will be freezed during training.
        "weight_decay": 0.01,
    },
    {
        "params": [p for n, p in encoder.glow.named_parameters()  \
                   if any(nd in n for nd in no_decay)],
        "weight_decay": 0,
    },
]
optimizer = AdamWeightDecayOptimizer(
    params=optimizer_grouped_parameters,
    lr=1e-3,
    eps=1e-6,
)


for epoch in range(num_epochs):
    size = len(train_dataloader.dataset)
    for step, batch in enumerate(train_dataloader):
        optimizer.zero_grad()
        batch['input_ids'] = batch['input_ids'].squeeze(1)
        batch['attention_mask'] = batch['attention_mask'].squeeze(1)
        batch = batch.to(device)
        
        encoder.train()
        z, loss = encoder(batch['input_ids'],batch['attention_mask'],return_loss = True)
        loss.backward()
        optimizer.step()
        if step % 100 == 0:
            current = step * len(batch['input_ids'])
            print(f"loss: {loss:>7f}  [{current:>5d}/{size:>5d}]")
            encoder.eval()
            srcc = eval_srcc(dev_data, encoder)
            encoder.train()
            print(f"corrcoef_dev: {srcc:>4f}")
```
